pazuzu.go

```diff
@@ -159,5 +169,160 @@ func (p *Pazuzu) DockerBuild(name string) error {
 		return err
 	}
 
+	p.testDockerImage(name)
+
+	return nil
+}
+
+func (p *Pazuzu) dockerExec(ID string, cmd string) error {
+	execOpts := docker.CreateExecOptions{
+		Container:    ID,
+		AttachStdin:  false,
+		AttachStdout: true,
+		AttachStderr: true,
+		Cmd: []string{
+			"/bin/bash",
+			"-c",
+			cmd,
+		},
+		Tty: true,
+	}
+	exec, err := p.docker.CreateExec(execOpts)
+	if err != nil {
+		return err
+	}
+
+	var buf bytes.Buffer
+	var errBuf bytes.Buffer
+
+	startExecOpts := docker.StartExecOptions{
+		Detach:       false,
+		OutputStream: os.Stdout,
+		ErrorStream:  &errBuf,
+		RawTerminal:  true,
+		Tty:          true,
+	}
+
+	err = p.docker.StartExec(exec.ID, startExecOpts)
+	if err != nil {
+		return err
+	}
+
+	inspect, err := p.docker.InspectExec(exec.ID)
+	if err != nil {
+		return err
+	}
+
+	if inspect.ExitCode > 0 {
+		return fmt.Errorf("exit code %d: %s", inspect.ExitCode, buf.String())
+	}
+
+	return nil
+}
+
+func (p *Pazuzu) dockerStart(image string) (*docker.Container, error) {
+	var err error
+	p.docker, err = docker.NewClient(p.DockerEndpoint)
+	if err != nil {
+		return nil, err
+	}
+	opts := docker.CreateContainerOptions{
+		Config: &docker.Config{
+			Image: image,
+			Tty:   true,
+			Cmd: []string{
+				"/bin/sh",
+			},
+		},
+		HostConfig: &docker.HostConfig{
+			Binds: []string{
+				tempDir + ":" + mountPoint,
+			},
+		},
+	}
+
+	container, err := p.docker.CreateContainer(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := p.docker.StartContainer(container.ID, nil); err != nil {
+		return nil, err
+	}
+
+	return container, nil
+}
+
+func (p *Pazuzu) dockerStop(ID string) error {
+	if err := p.docker.StopContainer(ID, 1); err != nil {
+		return err
+	}
+
+	if err := p.docker.RemoveContainer(docker.RemoveContainerOptions{
+		ID: ID,
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Pazuzu) generateTestSpec(features []shared.Feature) error {
+	var buffer = bytes.NewBufferString("")
+	if err := shared.WriteTestSpec(buffer, features); err != nil {
+		return err
+	}
+	p.TestSpec = buffer.Bytes()
+	return nil
+}
+
+func (p *Pazuzu) testDockerImage(image string) error {
+	os.MkdirAll(tempDir, 0777)
+
+	if err := exec.Command(
+		"wget",
+		"https://github.com/sstephenson/bats/archive/master.zip",
+		"-O", tempDir+"master.zip").Run(); err != nil {
```

Comment:
Well, beauty is in the eye of the beholder. :P But I'm also fine with the concatenation, we are already using it in other places... :)